src/main/java/com/android/volley/AsyncRequestQueue.java

```diff
@@ -426,15 +437,17 @@ private void finishRequest(Request<?> mRequest, Response<?> response, boolean ca
     }
 
     /**
-     * This interface may be used by advanced applications to provide custom executors according to
+     * This class may be used by advanced applications to provide custom executors according to
      * their needs. Apps must create ExecutorServices dynamically given a blocking queue rather than
      * providing them directly so that Volley can provide a PriorityQueue which will prioritize
      * requests according to Request#getPriority.
      */
-    public interface ExecutorFactory {
```

Comment:
Why not keeping the interface?

Reply:
This is taken from the Android API guidelines - since we can't use default interface methods in Volley, abstract classes are preferable because it gives us the flexibility to add methods in later versions. Since it seems like there's clearly value in such flexibility (as we're adding a scheduled Executor in this very PR), it seemed like a good time to make it an abstract class as well. I can't think of a concrete example of something we'd be likely to add here now that the scheduled executor is here, but I also don't see much benefit to keeping this an interface since I can't imagine anybody needs multiple inheritance to implement it.